commit 7974f8ddbe9f9594abcf9aac951138c636d79a2a
Author: Volker Krause <vkrause@kde.org>
Date:   Sun Jul 30 15:02:48 2017 +0200

    Fix build on Linux without X11
    
    Reviewers: #frameworks
    
    Tags: #frameworks
    
    Differential Revision: https://phabricator.kde.org/D6992

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 82220d6..76f4b8a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,10 +19,11 @@ include(KDECMakeSettings)
 
 if (NOT APPLE)
     find_package(X11)
+    find_package(Wayland)
 endif()
 
 set(HAVE_X11 ${X11_FOUND})
-
+set(HAVE_WAYLAND ${Wayland_FOUND})
 
 include(GenerateExportHeader)
 include(ECMSetupVersion)
diff --git a/src/config-kcrash.h.cmake b/src/config-kcrash.h.cmake
index fa032a3..1d92bec 100644
--- a/src/config-kcrash.h.cmake
+++ b/src/config-kcrash.h.cmake
@@ -1,4 +1,5 @@
 #cmakedefine01 HAVE_X11
+#cmakedefine01 HAVE_WAYLAND
 #cmakedefine KCRASH_CORE_PATTERN_RAISE
 
 #define kde_socklen_t socklen_t
diff --git a/src/kcrash.cpp b/src/kcrash.cpp
index 830a036..778cd4f 100644
--- a/src/kcrash.cpp
+++ b/src/kcrash.cpp
@@ -265,6 +265,8 @@ static const char* displayEnvVarName_c()
     // Can't use QGuiApplication::platformName() here, there is no app instance.
 #if HAVE_X11
     return "DISPLAY";
+#elif HAVE_WAYLAND
+    return "WAYLAND_DISPLAY";
 #elif defined(Q_OS_OSX)
     return "MAC_DISPLAY";
 #elif defined(Q_OS_WIN)
